### ISSUE TITLE
Bug: Cypress Component testing on React logs console logs recursively 

### ISSUE THREAD

**Issue by RomelClavel** [archfz/cypress-terminal-report#232]
When running a component test suite on cypress with the default setup, any console event will be printed into console recursively until there is a `RangeError: Maximum call stack size exceeded`.

The conditions for this behavior to occur are the following:
- There need to be at least 2 component tests in the suite. The undesired behavior does not trigger if there is only one test being run.
- There needs to be a one or more console events (log/info/warn/error) in the test case itself or in any component that's being rendered. 

Here is a simple example of the test suite that leads to this behavior:

```
describe("Recursive logging on console events", () => {

  const Title = () => {
    return <h1>title</h1>;
  };

  it("Component test 1", () => {
    cy.mount(<Title />);
    console.warn("warning");
    cy.get("h2");
  });

  it("Component test 2", () => {
    cy.mount(<Title />);
    console.warn("warning");
    cy.get("h2");
  });
})
```

Currently the only way I have of mitigating this unwanted behavior is excluding the console events setting the [Collect Types](https://www.npmjs.com/package/cypress-terminal-report#optionscollecttypes) option in the Install Logs Collector. Like so:
```

require("cypress-terminal-report/src/installLogsCollector")({
  collectTypes: ["cy:log", "cy:xhr", "cy:request", "cy:intercept", "cy:command"]
});
```

**Comment by archfz** [comment]
Fixed in 5.3.11

**Comment by xander-planted** [comment]
I've just set up v7.2.0 in an existing project and am seeing a similar issue — with the default configuration, a previously-passing test is now failing with a `RangeError` but no log output.

```
     RangeError: Invalid string length

Because this error occurred during a `after each` hook we are skipping all of the remaining tests.
      at stringify (<anonymous>)
      at <unknown> (http://localhost:4002/__cypress/iframes/cypress%2Fe2e%2Fposts-report-spec.cy.js?browserFamily=chromium:223:19)
      at Array.map (<anonymous>)
      at $Cypress.onCommandInvocation (http://localhost:4002/__cypress/iframes/cypress%2Fe2e%2Fposts-report-spec.cy.js?browserFamily=chromium:214:22)
      at listener (http://localhost:4002/__cypress/runner/cypress_runner.js:142243:19)
      at arrayMap (http://localhost:4002/__cypress/runner/cypress_runner.js:18631:23)
      at map (http://localhost:4002/__cypress/runner/cypress_runner.js:27600:14)
      at $Cypress.<anonymous> (http://localhost:4002/__cypress/runner/cypress_runner.js:142245:14)
      at parent.<computed> [as emitMap] (http://localhost:4002/__cypress/runner/cypress_runner.js:142204:31)
      at cy.<computed> [as task] (http://localhost:4002/__cypress/runner/cypress_runner.js:140226:45)
```
